### PR TITLE
Add a watchdog script that runs separately and warns if the job fails

### PR DESCRIPTION
**Pull request by Uberi**
The watchdog script warns users if the analysis job fails.

It's a separate script because sometimes Spark errors (bugs?) can take down the entire IPython kernel with no exceptions. Also, it's less likely to get killed if there's an OOM error.

It runs on the same instance as the job runs on, is started by the bootstrap notebook, and then runs totally independently until it times out.

Alerts are sent to telemetry-alerts@mozilla.com, which people can subscribe to if they want these failure warnings.

### CODE REVIEW

**Comment by vitillo** [comment]
`watchdog.sh` is missing.

**Comment by Uberi** [comment]
Oops...

I tested this by changing the target to my email, and the timeout to a much smaller value. It will send an email using SES when the timeout is reached and the JOB_SUCCESS file doesn't exist.

File: crash_rate_aggregates/watchdog.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+# watchdog timer for crash aggregator - if the aggregator job doesn't successfully complete within 10 hours, send out warning emails to telemetry-alerts@mozilla.com
+# successful completion is marked by the existance of the file `crash_aggregates/JOB_SUCCESS`
+# intended to be used with run_crash_aggregator.ipynb
+
+SCRIPT_DIR=$(dirname "$(readlink -f -- "$0")") # get the directory this script is in
+
+subject='[TIMEOUT] Crash rate aggregates job'
+body='
+Crash rate aggregates job failed to complete within 10 hours. Check logs at https://analysis.telemetry.mozilla.org/cluster/schedule for details.
```
Comment (vitillo) [inline crash_rate_aggregates/watchdog.sh:11]: Please remove "Check logs...".

File: crash_rate_aggregates/watchdog.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+This is an automatically generated message from the crash aggregates watchdog. The source code is available at https://github.com/mozilla/moz-crash-rate-aggregates.
+'
+
+sleep 36000 # 10 hours * 60 minutes/hours * 60 seconds/minutes = 36000 seconds
```
Comment (vitillo) [inline crash_rate_aggregates/watchdog.sh:16]: What happens if the cluster dies before? If a Spark job throws an exception the cluster gets killed and this script is never invoked.

**Comment by Uberi** [comment]
https://github.com/mozilla/cerberus/pull/24 has a better way of doing things, though it increases the maximum theoretical delay before receiving an error to days.

**Pull request closed without merging**